src/pages/Browse/Browse.tsx

```diff
@@ -39,12 +56,49 @@ const Browse: React.FC = () => {
         }
     };
 
+    const handleDeleteRow = (id: number) => {
+        const updatedRows = tableRows.filter((row) => row.id !== id);
+        setTableRows(updatedRows);
+    };
+
+    const handleEditRow = (id: number) => {
+        const rowToEdit = tableRows.find((row) => row.id === id);
+        if (rowToEdit) {
+            setNewRow(rowToEdit);
+            handleDeleteRow(id);
+            setIsEditMode(true);
+        }
+    };
+
+    const handleSave = () => {
+        if (isEditMode) {
+            const updatedRows = [...tableRows, newRow];
+            setTableRows(updatedRows);
+        }
+        setNewRow({ id: 0, name: '', password: '', email: '' });
+        setIsEditMode(false);
+    };
+
+    const handleAddRow = () => {
+        if (isEditMode) {
+            handleSave();
+        } else {
+            const newRowWithId = {
+                ...newRow,
+                id: Math.floor(Math.random() * 1000000),
```

Comment:
add all the code related to users into a different file dont add directly to browse file

Reply:
and please the check the ui between buttons edit and delete